### PR TITLE
Make tax report lines tag names unique

### PR DESCRIPTION
**Pull request by oco-odoo**
[IMP] account: add constraint on tax report lines to avoid duplicate tag name in the same country

[FIX] l10n_ae: avoid duplicate tag names in tax report
    
Having duplicate tag names is bad for usability, as the user will ne be able to differenciate between tags when performing tax configuration.

### CODE REVIEW

File: addons/account/models/account.py [hunk 1]
```diff
@@ -149,6 +149,12 @@ def _validate_tags(self):
         if self.tag_ids and (len(neg_tags) !=1 or len(pos_tags) != 1):
             raise ValidationError(_("If tags are defined for a tax report line, only two are allowed on it: a positive and a negative one."))
 
+    @api.constrains('tag_name')
+    def _validate_tag_name_unicity(self):
+        other_lines_with_same_tag = self.env['account.tax.report.line'].search([('tag_name', '=', self.tag_name), ('id', '!=', self.id), ('country_id', '=', self.country_id.id)])
```
Comment (sswapnesh) [inline addons/account/models/account.py]:
```suggestion
        other_lines_with_same_tag = self.env['account.tax.report.line'].search_count([('tag_name', '=', self.tag_name), ('id', '!=', self.id), ('country_id', '=', self.country_id.id)])
```

Comment (oco-odoo) [reply addons/account/models/account.py]: Good point. Thanks!

File: addons/account/models/account.py [hunk 2]
```diff
@@ -149,6 +149,12 @@ def _validate_tags(self):
         if self.tag_ids and (len(neg_tags) !=1 or len(pos_tags) != 1):
             raise ValidationError(_("If tags are defined for a tax report line, only two are allowed on it: a positive and a negative one."))
 
+    @api.constrains('tag_name')
```
Comment (sswapnesh) [inline addons/account/models/account.py]:
```suggestion
    @api.constrains('tag_name', 'country_id')
```
So we can prevent this when someone changes Country on existing record. (IDK know the actual workflow but just saying based on code.)

Comment (oco-odoo) [reply addons/account/models/account.py]: country_id isn't supposed to change, but I agree adding it to the constraint is more consistent. Thank you for your comment !

**Comment by jco-odoo** [comment]
@nim-odoo @qdp-odoo Looks good to me for merging.

**Comment by oco-odoo** [comment]
@jco-odoo can you merge ? :-)

**Comment by jco-odoo** [comment]
@qdp-odoo Can you have a quick look at the changes in account for the uniqueness of the tag name?  To me and oco it is better to change this now in 12.3 to avoid further errors.

**Comment by qdp-odoo** [comment]
@robodoo r+ rebase-ff

**Comment by rdeodoo** [comment]
@qdp-odoo Not sure if you noticed but this one failed

File: addons/account/models/account.py [hunk 3]
```diff
@@ -149,6 +149,12 @@ def _validate_tags(self):
         if self.tag_ids and (len(neg_tags) !=1 or len(pos_tags) != 1):
             raise ValidationError(_("If tags are defined for a tax report line, only two are allowed on it: a positive and a negative one."))
 
+    @api.constrains('tag_name', 'country_id')
+    def _validate_tag_name_unicity(self):
```
Comment (jco-odoo) [inline addons/account/models/account.py]: @oco-odoo Should check if tag_name is not False here.

**Comment by smetl** [comment]
@robodoo delegate+

**Comment by sswapnesh** [comment]
This one also needs @robodoo r+ again

**Comment by oco-odoo** [comment]
@sswapnesh Yup, I'm only delaying it a little bit before some other PR is ready ;) If we merge this one first, it will a few l10n modules uninstallable, and we don't want that

**Comment by oco-odoo** [comment]
robodoo r+

**Comment by oco-odoo** [comment]
robodoo retry

**Pull request closed without merging**